core/src/test-core/java/com/radixdlt/environment/deterministic/network/ControlledDispatcher.java

```diff
@@ -102,15 +105,16 @@ private static long addTimeNoOverflow(long a, long b) {
   }
 
   @Override
-  public <T> EventDispatcher<T> getDispatcher(Class<T> eventClass) {
+  public <T extends CoreEvent> EventDispatcher<T> getDispatcher(Class<T> eventClass) {
```

Comment:
LocalEvent

Reply:
Unfortunately, this change clashes with this piece of code ([Dispatchers.java](https://github.com/radixdlt/babylon-node/blob/4e79c00a70cd7c4ed06b77c587c40d6ee34e5e77/core/src/main/java/com/radixdlt/environment/Dispatchers.java#L162)):
```
    @Override
    public RemoteEventDispatcher<NodeId, T> get() {
      var remoteDispatcher = environmentProvider.get().getRemoteDispatcher(messageType);
      var localDispatcher = environmentProvider.get().getDispatcher(messageType);
```
messageType has RemoteEvent bound, while .getDisplatcher() requires LocalEvent. I have two variants: 
- Make CoreEvent as a type bound for .getDispatcher() return value
- Set type bound to RemoteEvent & LocalEvent and make all RemoteEvent implementations to implement LocalEvent too.
The latter variant looks like it defies the purpose of the type separation, so I decided to lower the type bound to the common CoreEvent type.

Reply:
Hmh, if we're dispatching a RemoteEvent to ourselves that means it must also be a LocalEvent. Wdyt about doing a runtime check and a cast or throw if this isn't a local event.

Reply:
This is a good idea, but I think that such rework is sensibly outside the scope of the ticket, which is not meant to change behavior. I think it makes sense to perform this change (as well as cleanup of remaining CoreEvent and fixing LedgerExtension issue) in the separate ticket/PR.